### ISSUE TITLE
Import for multilingual matrix text field overwrites other existing language specific content

### ISSUE THREAD

**Issue by timbertens** [craftcms/feed-me#634]
## Description
I have a section with a matrix field 'content builder', one of the available block is a redactor text field.  The site is multi language (3 languages, nl / fr / en) and we are setting up an import for that section using feedme.  The import of multilingual content in regular fields (not in matrix, but just the title, slug, 'standalone' textfield, etc) works fine, but on importing content into the redactor block in the matrix field, the content from the last import overwrites the previous imported language specific content from that field, in this case we end up with English content in the Dutch and French version of the entry. 

### Application Info

PHP version	7.3.1
OS version	Linux 4.15.0-45-generic
Database driver & version	MySQL 5.7.25
Image driver & version	GD 7.3.1
Craft edition & version	Craft Pro 3.3.16.3
Yii version	2.0.30
Twig version	2.12.2
Guzzle version	6.4.1
Imagine version	1.2.3-dev

### Plugins
Activity Log	1.0.1
CodeMirror	1.0.1
CP Field Inspect	1.0.7
Do It Yourself widget	2.0.1
Feed Me	4.1.2
Field Manager	2.1.1
GeoField	1.0.0
Imager	v2.3.0
Incognito Field	1.1.1.1
Matrix Toolbar	1.0.6
Migration Assistant	3.1.5
Redactor	2.4.0
SEOMate	1.1.1
Site Switcher	2.1.0
Stamp	2.1.0
Table Maker	2.0.1
Two-Factor Authentication	2.3.0
Typed link field	1.0.19
Wheel Form	2.3.4
Wheelform Helper	1.0.0
Wordsmith	3.1.1

**Comment by olivierbon** [comment]
@timbertens What setting do you have in
Settings → Fields → yourMatrixField → yourBlock → yourRedactorField → Translation Method?

**Comment by timbertens** [comment]
That is 'Translate for each language'

**Comment by timbertens** [comment]
Does anybody know if this is a general issue?

**Comment by bencroker** [comment]
Have you tried setting the propagation method to `Only save blocks to the site they were created in` for the matrix field?

**Comment by brandonkelly** [comment]
This is a duplicate of #463 

tl;dr - Feed Me doesn’t have a way of mapping your incoming Matrix block data with the existing blocks, so it will just wipe out the existing blocks and replace them with what’s in the feed.

The solution, as @bencroker pointed out, is you can set the Matrix field’s Propagation Method setting based on the behavior you want. For example if you are importing language-specific feeds, then set the Propagation Method to “Save blocks to other sites with the same language”. Then when importing the `de` feed, only blocks in sites with Language set to German will be deleted and re-imported.

Note that, up until the next release (will be 3.3.18), changing the Propagation Method setting will result in data loss, as it will wipe out all block content that doesn’t match the new setting value. I just fixed that behavior in craftcms/cms@daa4f2119cee2676cbc60d584da4203a2c1c0587, so now Craft will duplicate your existing Matrix blocks for sites where content is going to be deleted. I’d recommend either waiting for 3.3.18 to be released before changing the setting, or you can get that fix right away by editing your `craftcms/cms` requirement in `composer.json`:

```json
"require": {
  "craftcms/cms": "dev-develop#daa4f2119cee2676cbc60d584da4203a2c1c0587 as 3.3.17",
  "...": "..."
}
```

(Then run `composer update`.)

**Comment by bencroker** [comment]
That's great @brandonkelly!
